### PR TITLE
Totally new way to build Docker images

### PR DESCRIPTION
**Pull request by gwynne**
I offer, for some reason, this incredibly overbuilt setup that uses a Swift script (packaged as a full-blown executable) to calculate and build a few image permutations.

If nothing else, this is hopefully an interesting example of the weird things one could possibly do with `ConsoleKit` 😅

I will wait for review on this, especially on the set of images it builds, before actually pushing any of the built images...

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2018 Qutheory, LLC
```
Comment (tanner0101) [inline LICENSE]: 2020?

File: main.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ARG SWIFT_BASE_IMAGE
```
Comment (tanner0101) [inline main.Dockerfile]: Should this be called `swift.Dockerfile`?

**Review by tanner0101 — APPROVED** [review]
This is amazing :o

File: main.Dockerfile (hunk 2 above)
Comment (gwynne) [reply main.Dockerfile]: Yep and done :)

File: LICENSE (hunk 1 above)
Comment (gwynne) [reply LICENSE]: Done

**Pull request merged**